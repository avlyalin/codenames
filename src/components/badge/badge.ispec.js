/* eslint no-undef: 0 */

const customConfig = {
  failureThreshold: 0.01,
  failureThresholdType: 'percent',
};

describe('<Badge /> visually looks correct', () => {
  test('color: default', async () => {
    await page.goto(
      'http://localhost:9009/iframe.html?path=/story/components-badge--default',
    );
    const badge = await page.$('[data-testid=badge]');
    const image = await badge.screenshot();
    expect(image).toMatchImageSnapshot(customConfig);
  });
  test('color: blue', async () => {
    await page.goto(
      'http://localhost:9009/iframe.html?path=/story/components-badge--blue',
    );
    const badge = await page.$('[data-testid=badge]');
    const image = await badge.screenshot();
    expect(image).toMatchImageSnapshot(customConfig);
  });
  test('color: red', async () => {
    await page.goto(
      'http://localhost:9009/iframe.html?path=/story/components-badge--red',
    );
    const badge = await page.$('[data-testid=badge]');
    const image = await badge.screenshot();
    expect(image).toMatchImageSnapshot(customConfig);
  });
});
