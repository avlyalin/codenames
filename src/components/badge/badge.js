import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

const Badge = React.forwardRef(function Badge(props, ref) {
  const { rounded = true, color = 'default', children } = props;
  return (
    <span
      ref={ref}
      className={classnames(
        'inline-block px-2 py-1 leading-none text-center text-sm align-middle',
        {
          'rounded-xl': rounded,
          'bg-gray-300': color === 'default',
          'bg-red-100 text-white': color === 'red',
          'bg-blue-100 text-white': color === 'blue',
        },
      )}
    >
      {children}
    </span>
  );
});

Badge.propTypes = {
  rounded: PropTypes.bool,
  color: PropTypes.oneOf(['default', 'blue', 'red']),
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
};

export { Badge };
