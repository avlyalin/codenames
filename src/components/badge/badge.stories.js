import React from 'react';
import { withKnobs, boolean, radios } from '@storybook/addon-knobs';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { containerDecorator } from '_storybook/container';
import { Badge } from './badge';

// eslint-disable-next-line import/no-default-export
export default {
  title: 'Badge',
  decorators: [containerDecorator, withKnobs],
};

export const common = () => {
  const rounded = boolean('Rounded', true);
  const color = radios(
    'Color',
    { default: 'default', blue: 'blue', red: 'red' },
    'default',
  );
  return (
    <Badge color={color} rounded={rounded}>
      <FontAwesomeIcon icon="crown" size={'sm'} /> Капитан
    </Badge>
  );
};
