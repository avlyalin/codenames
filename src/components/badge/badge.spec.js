import React from 'react';
import { render } from '@testing-library/react';
import { Badge } from './badge';

describe('<Badge/>', () => {
  test('should render rounded gray badge', () => {
    const { container } = render(<Badge />);
    const badge = container.firstChild;
    expect(badge).toMatchSnapshot();
  });
  test('should render red badge with square borders', () => {
    const { container } = render(<Badge color="red" rounded={false} />);
    const badge = container.firstChild;
    expect(badge).toMatchSnapshot();
  });
});
