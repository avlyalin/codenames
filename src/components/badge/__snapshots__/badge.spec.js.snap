// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge/> should render red badge with square borders 1`] = `
<span
  class="inline-block px-2 py-1 leading-none text-center text-sm align-middle bg-red-100 text-white"
/>
`;

exports[`<Badge/> should render rounded gray badge 1`] = `
<span
  class="inline-block px-2 py-1 leading-none text-center text-sm align-middle rounded-xl bg-gray-300"
/>
`;
